src/coreclr/src/System.Private.CoreLib/src/System/Diagnostics/Eventing/EventPipe.CoreCLR.cs

```diff
@@ -0,0 +1,64 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+using System.Collections.Generic;
+using System.Runtime.CompilerServices;
+using System.Runtime.InteropServices;
+using System.Threading;
+
+#if FEATURE_PERFTRACING
+
+namespace System.Diagnostics.Tracing
+{
+    internal static partial class EventPipeInternal
+    {
+        //
+        // These PInvokes are used by the configuration APIs to interact with EventPipe.
+        //
+        [DllImport(RuntimeHelpers.QCall, CharSet = CharSet.Unicode)]
```

Comment:
We have been keeping all internal VM calls definitions in the runtime-specific CoreLib sources to keep things uniform.

> allow the same attribute to be used by both

These attributes have different meaning. InternalCall does not switch to preeemptive mode. DllImport does switch to preemptive mode.

Reply:
> We have been keeping all internal VM calls definitions in the runtime-specific CoreLib sources to keep things uniform.

I'm happy to follow suit then : )

> These attributes have different meaning. InternalCall does not switch to preeemptive mode. DllImport does switch to preemptive mode.

In an ideal world wouldn't we want the runtime code to determine whether the transition happens as its own implementation detail rather than forcing the distinction into the managed contract? I'm fine if we decide the benefits of sprucing up the code don't justify the work to do it though.

Reply:
I believe we will revisit this going forward when more of the native EventPipe code gets into Mono runtime as well as getting things over to mobile devices. We might very well end up moving into a loadable module (p/invoke) in the end due to size restrictions on mobile devices, trying to minimize size by excluding features, even not not including managed parts at all used by mobile devices, but before we get there, lets stick to icalls and keep it included as part of runtime to reduce complexity.